shell/platform/darwin/macos/framework/Source/FlutterViewController.mm

```diff
@@ -427,12 +429,23 @@ - (void)onPreEngineRestart {
   [self initializeKeyboard];
 }
 
+- (void)attachToEngine:(nonnull FlutterEngine*)engine withId:(uint64_t)viewId {
+  NSAssert(_engine == nil && _id == 0, @"Already attached to an engine, engine %@ ID %llu.",
+           _engine, _id);
+  _engine = engine;
+  _id = viewId;
+}
+
+- (void)detachFromEngine {
+  NSAssert(_engine != nil && _id != 0, @"Not attached to an engine, engine %@ ID %llu.", _engine,
+           _id);
+  _engine = nil;
+  _id = 0;
```

Comment:
Ah, are you shifting the default so 0 becomes available to denote "not attached"? Should we use a constant for this? Might also be worthwhile to mention that in the doc on the "id" property, i.e. that accessing the ID is only valid when an engine is attached? Maybe we can even assert if somebody tries to access the ID while no engine is attached.

Reply:
Also, if we make accessing "ID" invalid when no engine is attached, do we really need to reserve a value for that case? It feels a little odd to me that view IDs don't include zero...

Reply:
Currently the `id` property is a normal property, neither getter or setter overridden, which I think it's good simplicity. Technically we don't need to a special value for an invalid ID value, we can instead tell the user to not trust the ID value if the FVC is unattached. Do you prefer this way?

Reply:
I think I would prefer that over another magic value. We should document that, though. And I would still argue for properly asserting if someone accesses the ID while the controller is detached (AFAIK, currently, ObjC just generates the getter/setter, making those explicit, shouldn't be that complex).

Reply:
Done. Reading the ID now asserts that the engine is attached.